### ISSUE TITLE
FFTJet* modules still needed?

### ISSUE THREAD

**Issue by Dr15Jones** [cms-sw/cmssw#35578]
The FFTJetProducer, FFTJetPileupProcessor, FFTJetCorrectorDBReader,  and FFTJetCorrectionProducer modules are thread unsafe, do not use esConsumes and still inherit from the legacy modules. A quick check on FFTJetProducer and FFTJetCorrectionProducer did not show up any instances of configurations using the modules.

**Comment by slava77** [comment]
@igv4321 
please take a note of this issue and perhaps comment.
One option is to modernize the code, the other is to remove.

@Dr15Jones is this urgent for the core group?

**Comment by makortel** [comment]
> is this urgent for the core group?

Since they appear to be unused, these are not urgent, but they would be good to be addressed in some way within the next months.

**Comment by Dr15Jones** [comment]
@slava77 for perspective, the package and things using it are generating something like 400+ warnings in the CMSDEPRECATED IB, which is more than 10% of all warnings.

**Comment by makortel** [comment]
@igv4321 Would you have any news on these modules? The `esConsumes()` is now enforced at run time (since CMSSW_12_3_X_2021-12-13-1100).

**Comment by igv4321** [comment]
The plan is still to fix them before the end of the year.

Igor

**Comment by igv4321** [comment]
Sorry for the delay. Will need a few more days to fix this.

Regards,
Igor

**Comment by makortel** [comment]
@igv4321 Any news?

**Comment by igv4321** [comment]
Dear Matti,

Sorry for being really slow about fixing these. I should be
able to take care of these modules over the spring break
(by March 20).

Regards,
Igor

**Comment by jpata** [comment]
kind ping @igv4321

**Comment by igv4321** [comment]
Some fixes are provided in the PR #37999. Sorry for slow response on this.

**Comment by jpata** [comment]
+reconstruction
- the FFTJet modules were migrated to esConsumes in https://github.com/cms-sw/cmssw/pull/37999, so they can now in principle run
- it was stated that there are plans to revive studies with them, even though they are currently not running in any workflow

**Comment by makortel** [comment]
> * the FFTJet modules were migrated to esConsumes in [Changing FFTJet modules to stream modules #37999](https://github.com/cms-sw/cmssw/pull/37999), so they can now in principle run

#37999 changed only some EDModules from legacy to stream, but did not migrate anything to esConsumes. In CMSSW_12_5_X_2022-05-20-2300, that includes #37999, the CMSDEPRECATED_X build still reports 417 warnings for `JetMETCorrections/FFTJetModules` and `JetMETCorrections/FFTJetObjects`. On a quick look majority of those warnings are for esConsumes, but there were some legacy EDModules as well.

I'd ask to reopen this issue.

**Comment by igv4321** [comment]
@makortel  How do I turn these messages on? I don't see any warning messages with the usual "scram b" command.

**Comment by makortel** [comment]
I just tried the following in 12_5_0_pre1
```
git cms-addpkg JetMETCorrections/FFTJetModules JetMETCorrections/FFTJetObjects
scram b
```
and saw the 417 warnings.

**Comment by makortel** [comment]
Did the bot close this because of the earlier "please close" message even if @perrotta reopened this issue? Do we need to make the bot to recognize reopening?

**Comment by smuzaffar** [comment]
yes @makortel bot closed it due to https://github.com/cms-sw/cmssw/issues/35578#issuecomment-1134490302

**Comment by makortel** [comment]
Hi @igv4321, would you have any update on schedule to get `FFTJetModules` and `FFTJetObjects` migrated?

**Comment by igv4321** [comment]
@makortel This issue should be fixed by PR #38932. Sorry for the delay.

**Comment by makortel** [comment]
Thanks @igv4321!
